docs-client/src/components/KeyValueTable/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { Input } from '@material-ui/core';
+import React, { Dispatch, SetStateAction, useContext } from 'react';
+
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import { Row, ValueListContext } from '../KeyValueEditor/valueListContext';
+
+interface RowProps {
+  row: Row;
+  onRowChange: (index: number, key: string, value: string) => void;
+}
+
+const KeyValueTableRow: React.FunctionComponent<RowProps> = ({
+  row,
+  onRowChange,
+}) => {
+  const onChange = (rkey: string, rvalue: string) => {
+    onRowChange(row.index, rkey, rvalue);
+  };
+
+  return (
+    <TableRow key={row.index}>
+      <TableCell>
+        <Input
+          value={row.key}
+          onChange={(e) => onChange('key', e.target.value)}
+        />
+      </TableCell>
+      <TableCell>
+        <Input
+          defaultValue={row.value}
+          onChange={(e) => onChange('value', e.target.value)}
+        />
+      </TableCell>
+    </TableRow>
+  );
+};
+
+interface KeyValueTableProps {
+  defaultKeyValueList?: Row[];
+}
+
+const KeyValueTable: React.FunctionComponent<KeyValueTableProps> = ({}) => {
+  const resultArr:
+    | [Row[], Dispatch<SetStateAction<Row[]>>]
+    | undefined = useContext(ValueListContext);
+
+  if (!resultArr) throw new Error('KeyValueTable : RowList가 없습니다.');
+
+  const [rowList, setRowList] = resultArr;
+
+  const onRowChange = (index: number, name: string, value: string) => {
+    if (rowList) {
+      setRowList(
+        rowList.map((row, i) =>
+          i === index ? { ...row, [name]: value } : row,
+        ),
+      );
+    }
+  };
+  // tslint:disable-next-line
+  console.log(rowList);
+  return (
+    <Table>
+      <TableHead>
+        <TableRow>
+          <TableCell>KEY</TableCell>
+          <TableCell>VALUE</TableCell>
```

Comment:
An HTTP header has a name and values, so `Name` and `Value`?

Reply:
> I'm curious if we're going to have a 'delete' button and other bells and whistles as well.
> 
> Or would it be better using 3rd party table edit control? What do you think?

yes It need some features, Finding library is also good way.
(but it may hard to find because there's a lot of conditions like React && TS && Material UI && good to use)
So I consider it

Reply:
> An HTTP header has a name and values, so `Name` and `Value`?

I agree